src/components/icon/index.tsx

```diff
@@ -83,7 +130,21 @@ const Icon = forwardRef((props: Props, ref: any) => {
     return renderImage();
   }
 
-  return isSvg(source) ? renderSvg() : renderImage();
+  return (
+    <View>
+      {isSvg(source) ? renderSvg() : renderImage()}
+      {shouldShowBadge && (
+        <Badge
+          pointerEvents={'none'}
+          {...badgeProps}
+          label={getBadgeLabel()}
```

Comment:
why not just pass the label from the badgeProps? Why the getter?

Reply:
you are right, I used the logic from the private repo - `BadgeButton` logic.